public/app/features/templating/editor_ctrl.ts

```diff
@@ -155,6 +155,8 @@ export class VariableEditorCtrl {
           return { text: ds.meta.name, value: ds.meta.id };
         })
         .value();
+
+      $scope.currentDatasource = null;
```

Comment:
If you have edited/added a query variable and adding a new variable this change will make the variable editor reset properly so that no datasource is selected and by that the query editor is not displayed until a datasource have been selected.

Reply:
👍